compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -257,26 +258,27 @@ shakeRun IdeState{shakeExtras=ShakeExtras{..}, ..} acts = modifyVar shakeAbort $
 -- | Use the last stale value, if it's ever been computed.
 useStale
     :: IdeRule k v
-    => IdeState -> k -> FilePath -> IO (Maybe v)
-useStale IdeState{shakeExtras=ShakeExtras{state}} k fp =
+    => ShakeExtras -> k -> FilePath -> IO (Maybe v)
```

Comment:
Why did this change type? Why not get the ShakeExtras from inside IdeState? I imagine ShakeExtras won't be exposed publicly, but IdeState will.